src/andromede/simulation/linearize.py

```diff
@@ -159,29 +169,28 @@ def division(self, node: DivisionNode) -> LinearExpressionData:
             t.coefficient /= divider
         return actual_expr
 
-    # TODO: linear expressions should be re-usable for different timesteps and scenarios
-    timestep: int
-    scenario: int
-    value_provider: Optional[ParameterGetter] = None
-
-    def _get_timestep(self, time_index: TimeIndex) -> int:
+    def _get_timestep(self, time_index: TimeIndex) -> Optional[int]:
         if isinstance(time_index, TimeShift):
+            if self.timestep is None:
+                raise ValueError("Cannot shift a time-independent expression.")
```

Comment:
Is there cases where it would be necessary to return `None` rather than raising an exception ?

Reply:
We should have a "node type checker" visitor to actually check that at various points of the transformations we only have certains node types.
Would be nice to do that in another PR.